src/ManualInput/Input.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { forwardRef, useImperativeHandle, useRef, useState } from 'react';
+import { StyleSheet, Text, TextInput, TouchableOpacity, View } from 'react-native';
+
+import { ManualInputDefault } from '../types';
+const obfusChars = ['/', ':'];
+type Props = {
+    defaultValue: ManualInputDefault;
+    onInputChange: (text: string) => void;
+    maxLength?: number;
+};
+export type FocusHandle = {
+    focus: () => void;
+};
+const Input = forwardRef<FocusHandle, Props>(({ defaultValue, onInputChange, maxLength }, ref) => {
+    const [value, setValue] = useState('');
+    const textInputRef = useRef<TextInput>(null);
+
+    useImperativeHandle(ref, () => ({
+        focus: () => {
+            textInputRef?.current?.focus();
+        },
+    }));
+    const minWidth = defaultValue === ManualInputDefault.LongDate ? 130 : 94;
+
+    const handleChange = (text: string) => {
+        setValue(text);
+        onInputChange(text);
+    };
+
+    const renderMaskedText = (text: string) => {
+        let val = defaultValue.split('');
+        let j = 0;
+        for (let i = 0; i < defaultValue.length; i++) {
+            if (obfusChars.includes(defaultValue[i])) {
+                val[i] = defaultValue[i];
+            } else {
+                val[i] = text[j] ?? defaultValue[i];
+                j++;
+            }
+        }
+        return <Text style={styles.maskedText}>{val.join('')}</Text>;
+    };
+
+    return (
+        <View style={styles.container}>
+            <TouchableOpacity
+                onPress={() => {
+                    textInputRef.current?.focus();
+                }}
```

Comment:
same function is used in `useImperativeHandler` can be extracted into function and mention here